pandas/core/arrays/interval.py

```diff
@@ -244,6 +245,19 @@ def _simple_new(
             )
             raise ValueError(msg)
 
+        from pandas.core.ops.array_ops import maybe_upcast_datetimelike_array
+
+        left = maybe_upcast_datetimelike_array(left)
+        right = extract_array(right, extract_numpy=True)
+        left = extract_array(left, extract_numpy=True)
+        right = maybe_upcast_datetimelike_array(right)
```

Comment:
can you order  these left then right. also the ordering of the function calls for left is differnt?